tests/php/Control/LoginSessionMiddlewareTest.php

```diff
@@ -0,0 +1,127 @@
+{
+    use HttpRequestMockBuilder;
+
+    protected $usesDatabase = true;
+
+    protected static $fixture_file = 'LoginSessionMiddlewareTest.yml';
+
+    public function testMiddlewareUpdatesLoginSession()
+    {
+        // log out
+        Security::setCurrentUser(null);
+
+        $session = new Session(['activeLoginSession' => 1]);
+        $request = $this->buildRequestMock('/', [], [], null, $session);
+        $request->method('getIP')->willReturn('192.168.0.1');
+
+        $middleware = new LoginSessionMiddleware(new Url('no-match'));
+        $next = false;
+        $middleware->process(
+            $request,
+            static function () use (&$next) {
+                $next = true;
+            }
+        );
+        $this->assertTrue($next, "Middleware does not do anything if not logged in");
+
+        // Then log in as member1
+        $member1 = $this->objFromFixture(Member::class, 'member1');
+        Security::setCurrentUser($member1);
+        DBDatetime::set_mock_now('2003-02-15 12:00:00');
+
+        $middleware = new LoginSessionMiddleware(new Url('no-match'));
+        $next = false;
+        $middleware->process(
+            $request,
+            static function () use (&$next) {
+                $next = true;
+            }
+        );
+
+        $loginSession = LoginSession::get()->byID(1);
+        $this->assertEquals(
+            $loginSession->LastAccessed,
+            '2003-02-15 12:00:00',
+            "Middleware updates last accessed of login session when logged in"
+        );
+        $this->assertEquals(
+            $loginSession->IPAddress,
+            '192.168.0.1',
+            "Middleware updates ip address of login session when logged in"
+        );
+        $this->assertEquals(
+            Security::getCurrentUser(),
+            $member1,
+            "Middleware keeps member logged in when session is valid"
+        );
+    }
+
+    public function testMiddlewareSessionRevoked()
+    {
+        $session = new Session(['activeLoginSession' => 1]);
+        $request = $this->buildRequestMock('/', [], [], null, $session);
+        $request->method('getIP')->willReturn('192.168.0.1');
+
+        // Log in as member1
+        $member1 = $this->objFromFixture(Member::class, 'member1');
+        Security::setCurrentUser($member1);
+
+        $loginSession = LoginSession::get()->byID(1);
+        $loginSession->delete();
+
+        $middleware = new LoginSessionMiddleware(new Url('no-match'));
+        $next = false;
+        $middleware->process(
+            $request,
+            static function () use (&$next) {
+                $next = true;
+            }
+        );
+
+        $this->assertNull(
+            Security::getCurrentUser(),
+            "Middleware logs user out if session is revoked"
+        );
+    }
+
+    public function testMiddlewareSessionWrongMember()
+    {
+        $session = new Session(['activeLoginSession' => 1]);
```

Comment:
You can retrieve the ID of the fixture programmatically, rather than hardcoding it:

```suggestion
        $sessionID = $this->idFromFixture(LoginSession::class, '1');
        $session = new Session(['activeLoginSession' => $sessionID]);
```